core/state.go

```diff
@@ -139,10 +139,11 @@ func (s *State) globalTrie(bucket db.Bucket, newTrie trie.NewTrieFunc) (*trie.Tr
 
 	// fetch root key
 	rootKeyDBKey := dbPrefix
-	var rootKey *trie.Key
+	var rootKey *trie.BitArray
 	err := s.txn.Get(rootKeyDBKey, func(val []byte) error {
-		rootKey = new(trie.Key)
-		return rootKey.UnmarshalBinary(val)
+		rootKey = new(trie.BitArray)
+		rootKey.UnmarshalBinary(val)
+		return nil
```

Comment:
The approach that you suggested wouldn't work because `rootKey` still need to be used in the caller function (i.e. `globalTrie`). I tried something like:

```
	var rootKey trie.BitArray
	err := s.txn.Get(rootKeyDBKey, func(val []byte) error {
		rootKey.UnmarshalBinary(val)
		return nil
	})
``` 

Tests failed though. I'm just gonna leave a TODO there and avoid breaking things for now.

Reply:
That said, the entire fetch rootKey thing shouldn't even be there in the first place.